header_test.go

```diff
@@ -728,3 +734,86 @@ func TestCommaDelimitedAddressLists(t *testing.T) {
 		}
 	}
 }
+
+func TestParseMediaType(t *testing.T) {
+	testCases := []struct {
+		label  string            // Test case label.
+		input  string            // Content type to parse.
+		mtype  string            // Expected media type returned.
+		params map[string]string // Expected params returned.
+	}{
+		{
+			label:  "basic filename",
+			input:  "text/html; name=index.html",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index.html"},
+		},
+		{
+			label:  "quoted filename",
+			input:  "text/html; name=\"index.html\"",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index.html"},
+		},
+		{
+			label:  "basic filename trailing separator",
+			input:  "text/html; name=index.html;",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index.html"},
+		},
+		{
+			label:  "quoted filename trailing separator",
+			input:  "text/html; name=\"index.html\";",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index.html"},
+		},
+		{
+			label:  "unclosed quoted filename",
+			input:  "text/html; name=\"index.html",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index.html"},
+		},
+		{
+			label:  "quoted filename with separator",
+			input:  "text/html; name=\"index;a.html\"",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index;a.html"},
+		},
+		{
+			label:  "quoted separator mid-string",
+			input:  "text/html; name=\"index;a.html\"; hash=8675309",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index;a.html", "hash": "8675309"},
+		},
+		{
+			label:  "quoted separator with escaped quote mid-string",
+			input:  "text/html; name=\"index\\\";a.html\"; hash=8675309",
+			mtype:  "text/html",
+			params: map[string]string{"name": "index\";a.html", "hash": "8675309"},
+		},
+	}
```

Comment:
The test cases, above, seem like what I would expect. I'd like @requaos to take a look with what he's seen in the wild, too.